### ISSUE TITLE
I2C function signature mistmatch in core vs mbos

### ISSUE THREAD

**Issue by pelikhan** [lancaster-university/codal-core#26]
The I2C function signatures in codal-core and codal-mbedos are different (uint16_t vs uint32_t). This breaks i2c in mbedos since all base class function are noops.

https://github.com/lancaster-university/codal-core/blob/master/inc/driver-models/I2C.h#L115

vs

https://github.com/lancaster-university/codal-mbedos/blob/master/inc/MbedI2C.h#L81

Is this intentional?

**Comment by jamesadevine** [comment]
@pelikhan True, we moved codal-mbed to use the new abstract driver implementation, but not mbed-os.

https://github.com/lancaster-university/codal-mbed/blob/master/source/MbedI2C.cpp

It means that you only need to implement about three functions, and the rest is handled by the higher level abstraction.

@finneyj was there a reason why we didn't move mbedos forward?
